File: LeetCode/0056/python/lc0056_test.py
import unittest

from lc0056 import Solution

class Test_merge(unittest.TestCase):
    def test(self):
        s = Solution()
        self.assertEqual(s.merge([[1, 3], [2, 6], [8, 10], [15, 18]]), [[1, 6], [8, 10], [15, 18]])
        self.assertEqual(s.merge([[1, 4], [4, 5]]), [[1, 5]])
        self.assertEqual(s.merge([[4, 7], [1, 4]]), [[1, 7]])

if __name__ == "__main__":
    unittest.main()

File: LeetCode/0056/golang/lc0056_test.go
package lc0056

import (
	"slices"
	"testing"
)

func IsEqual(lhs [][]int, rhs [][]int) bool {
	if len(lhs) != len(rhs) {
		return false
	}
	for i := range lhs {
		if !slices.Equal(lhs[i], rhs[i]) {
			return false
		}
	}
	return true
}

func Test_merge(t *testing.T) {
	testcases := []struct {
		intervals [][]int
		expected  [][]int
	}{
		{
			[][]int{{1, 3}, {2, 6}, {8, 10}, {15, 18}},
			[][]int{{1, 6}, {8, 10}, {15, 18}},
		},
		{
			[][]int{{1, 4}, {4, 5}},
			[][]int{{1, 5}},
		},
		{
			[][]int{{4, 7}, {1, 4}},
			[][]int{{1, 7}},
		},
		{
			[][]int{{2, 3}, {2, 2}, {3, 3}, {1, 3}, {5, 7}, {2, 2}, {4, 6}},
			[][]int{{1, 3}, {4, 7}},
		},
	}

	for i, tc := range testcases {
		actual := merge(tc.intervals)
		if !IsEqual(actual, tc.expected) {
			t.Errorf("Testcase merge#%02d (%v) failed: want %v, got %v",
				i+1, tc.intervals, tc.expected, actual)
		}
	}
}
